Readme.md
# vcc

Extension to clear Varnish cache on demand.

## Extension settings

**Varnish Server**

Comma separated list of IP addresses of your Varnish server(s).

**HTTP ban method**

HTTP method to send to the Varnish server(s) to invalidate cache. Default: *BAN*

**HTTP protocol**

The HTTP protocol to use for the HTTP method.

**Cache handling**

- *Disabled*: No request is sent to the Varnish server(s).
- *Automatic*: Requests are sent immediately after a record was saved.
- *Manuel*: An icon is added to the toolbar and requests have to be triggered manually.

**Strip slash**

Strip appended slash in requested url to be able to adjust behaviour in Varnish vcl configuration.

**Support index.php script**

If enabled a request for index.php?id= script is sent as well.

**Logging mode**

- *Disabled*: No log information are stored in database.
- *Minimal*: One log information for each request is stored in database.
- *Debug*: Multiple log information for each request are stored in database.

**Maximum age of log entries**

Sets the maximum age of log entries in days.


ChangeLog
2019-08-30         [RELEASE] Release of vcc 4.1.0 (Dirk Wenzel)
2019-08-30 60a5ce8 Merged bug fixes and features from @meisterkaiser (Thomas Kaiser)

2018-03-13         [RELEASE] Release of vcc 3.1.0 (Nicole Cordes)
2018-03-13 01ebc2f [TASK] Raise compatibility to TYPO3 8.7 (Nicole Cordes)
2017-06-21 619680a [BUGFIX] Enforce resetting PageRenderer->backPath in TYPO3 < 8 (Nicole Cordes)
2017-06-20 0d32759 [TASK] Add Readme.md (Nicole Cordes)
2016-12-05 64c374a [TASK] Raise compatibility to TYPO3 8.5 (Nicole Cordes)

2016-11-18         [RELEASE] Release of vcc 3.0.1 (Nicole Cordes)
2016-11-18 eec2e20 [HOTFIX] Care about PageRenderer->backPath (Nicole Cordes)

2016-10-13         [RELEASE] Release of vcc 3.0.0 (Nicole Cordes)
2016-07-21 55fd990 [TASK] Use correct function call (Nicole Cordes)
2016-03-03 9b15164 [TASK] Use ::class instead of strings (Nicole Cordes)
2016-03-03 6b4dbae [TASK] Switch to psr-1/psr-2 coding standards (Nicole Cordes)
2016-03-03 8e18d36 [TASK] Switch to ModuleTemplate -> ButtonBar hook for icon processing (Nicole Cordes)
2016-03-03 f35ee85 [TASK] Use IconRegistry for icon registration (Nicole Cordes)
2016-03-03 9cb44e3 [BUGFIX] Remove duplicated hsc for CommunicationService notifications (Nicole Cordes)
2016-03-03 93cf041 [BUGFIX] Remove deprecated call to $GLOBALS['TSFE']->getPageRenderer() (Nicole Cordes)
2016-03-03 919ff78 [TASK] Set dependency to TYPO3 7 (Nicole Cordes)
2015-12-24 e8366cf [TASK] Add psr-4 autoload information (Nicole Cordes)
2015-08-27 2dca6c6 [BUGFIX] Prevent new pages from being cleared (Nicole Cordes)
2015-08-27 0576e3b [TASK] Implement new hook function to manipulate database record (Nicole Cordes)
2015-08-27 7a7736e [!!!][TASK] Streamline CommunicationServiceHookInterface (Nicole Cordes)
2015-08-27 601b1eb [TASK] Implement JavaScript notification API (Nicole Cordes)
2015-08-27 b8ba251 [TASK] Add public getter (Nicole Cordes)
2015-08-26 370fd6e [TASK] Add namespace (Nicole Cordes)
2015-08-26 8a85e69 [FEATURE] Add new hook to change the processed url (Nicole Cordes)
2015-08-26 bffb040 [TASK] Rename pid parameter to pageId (Nicole Cordes)
2015-08-24 c2e44d5 [TASK] Remove manual class loading (Nicole Cordes)
2015-08-24 5b4fccc [TASK] Adjust script entry point check (Nicole Cordes)
2015-08-24 a17b271 [TASK] Remove old XClass code (Nicole Cordes)
2015-08-24 f60e7ac [TASK] Replace deprecated class TSpagegen (Nicole Cordes)
2015-08-24 e5154e8 [TASK] Replace deprecated class t3lib_pageSelect (Nicole Cordes)
2015-08-24 8c92d21 [TASK] Replace deprecated class t3lib_TimeTrackNull (Nicole Cordes)
2015-08-24 6711dab [TASK] Replace deprecated class t3lib_TCEmain (Nicole Cordes)
2015-08-24 bdba4ba [TASK] Replace deprecated class t3lib_PageRenderer (Nicole Cordes)
2015-08-24 2422b2c [TASK] Replace deprecated class t3lib_iconWorks (Nicole Cordes)
2015-08-24 2d3f64d [TASK] Replace deprecated class t3lib_parsehtml (Nicole Cordes)
2015-08-24 356663c [TASK] Replace deprecated class t3lib_BEfunc (Nicole Cordes)
2015-08-24 a67f076 [TASK] Replace deprecated class tslib_fe (Nicole Cordes)
2015-08-24 0355224 [TASK] Replace deprecated class tslib_cObj (Nicole Cordes)
2015-08-24 b26c991 [TASK] Remove legacy code for minor TYPO3 versions (Nicole Cordes)
2015-08-24 3ae2dcc [TASK] Replace deprecated class t3lib_FlashMessage (Nicole Cordes)
2015-08-24 adb11e3 [TASK] Replace deprecated class t3lib_div (Nicole Cordes)
2015-08-24 ab4e791 [TASK] Replace deprecated class t3lib_Singleton (Nicole Cordes)
2015-08-24 cf5c365 [TASK] Replace deprecated class t3lib_SpriteManager (Nicole Cordes)
2015-08-24 edcf4f7 [TASK] Replace deprecated class t3lib_extMgm (Nicole Cordes)

2016-02-17         [RELEASE] Release of vcc 2.0.4 (Nicole Cordes)
2016-02-17 e789be1 [HOTFIX] Prevent any pageNotFound handling (Nicole Cordes)

2016-02-02         [RELEASE] Release of vcc 2.0.3 (Nicole Cordes)
2016-02-02 2d99e2b [HOTFIX] Init fe_user and usergroups for proper page access check (Nicole Cordes)

2015-07-30         [RELEASE] Release of vcc 2.0.2 (Nicole Cordes)
2015-07-30 8d593b3 [HOTFIX] Get page access permissions according to table (Nicole Cordes)

2015-07-09         [RELEASE] Release of vcc 2.0.1 (Nicole Cordes)
2015-07-09 f115bb3 [HOTFIX] Catch fatal errors during frontend initialization (Nicole Cordes)

2015-03-04         [RELEASE] Release of vcc 2.0.0 (Nicole Cordes)
2014-12-16 ca98f9a [BUGFIX] Prevent start url from stripping (Nicole Cordes)
2014-12-16 88df44b [BUGFIX] Get correct page id for "pages" records (Nicole Cordes)
2014-12-03 9b4e423 [FEATURE] Add disabled mode to use api only (Nicole Cordes)
2014-11-26 2f4d372 [!!!] [FEATURE] Add parameter to hook interface functions (Nicole Cordes)
2014-11-26 e857be7 [BUGFIX] Lowercase hook interface class name in ext_autoload.php (Nicole Cordes)
2014-11-26 1a63f19 [TASK] Remove unnecessary header (Nicole Cordes)
2014-11-26 98f9d50 [TASK] Improve default.vcl for basic version (Nicole Cordes)
2014-11-26 befaac7 [TASK] Rename hook classes with meaningful class names (Nicole Cordes)
2014-11-20 11533f7 [TASK] Clean up PHPDoc comments (Nicole Cordes)
2014-11-19 41d3faf [FEATURE] Add different logging modes (Nicole Cordes)
2014-09-23 5d3407a [BUGFIX] Retrieve correct uid for new records (Nicole Cordes)
2014-08-22 d2eccc5 [TASK] Adjust example vcl to work with automatic mode (Nicole Cordes)
2014-08-22 3414585 [FEATURE] Add automatic cache clear mode (Nicole Cordes)

2013-11-22         [RELEASE] Release of vcc 1.0.2 (Nicole Cordes)
2013-11-22 1ddaf66 [TASK] Add old manual with link to rest documentation (Nicole Cordes)
2013-11-22 c40b93d [TASK] Rename administration section in documentation (Nicole Cordes)
2013-11-21 9ff1d33 [TASK] Restructure documentation (Nicole Cordes)
2013-11-20 03c9a09 [BUGFIX] Improve code quality according to metrics.typo3.org (Nicole Cordes)

2013-11-18         [RELEASE] Release of vcc 1.0.1 (Nicole Cordes)
2013-11-18 9f08360 [TASK] Add documentation (Nicole Cordes)


ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "vcc".
 *
 * Auto generated 13-03-2018 14:47
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = array (
  'title' => 'Varnish Cache Control',
  'description' => 'Varnish cache tools for TYPO3. Supports ESI tags and cache control.',
  'author' => 'Nicole Cordes',
  'author_email' => 'cordes@cps-it.de',
  'author_company' => 'CPS-IT',
  'category' => 'module',
  'constraints' => 
  array (
    'depends' => 
    array (
      'typo3' => '7.6.0-8.7.99',
    ),
    'conflicts' => 
    array (
    ),
    'suggests' => 
    array (
        'scheduler' => 's'
    ),
  ),
  'state' => 'stable',
  'version' => '4.1.0',
  'clearCacheOnLoad' => 0,
  'createDirs' => '',
  'uploadfolder' => 0,
  '_md5_values_when_last_written' => 'a:35:{s:9:"ChangeLog";s:4:"9de5";s:9:"Readme.md";s:4:"78c9";s:21:"ext_conf_template.txt";s:4:"0425";s:12:"ext_icon.gif";s:4:"7736";s:17:"ext_localconf.php";s:4:"565d";s:14:"ext_tables.php";s:4:"08da";s:14:"ext_tables.sql";s:4:"685c";s:37:"Classes/Hooks/AbstractVarnishHook.php";s:4:"83c5";s:36:"Classes/Hooks/ClearCacheIconHook.php";s:4:"fcbb";s:43:"Classes/Hooks/ClearCachePostProcessHook.php";s:4:"2dd4";s:44:"Classes/Hooks/RecordSavedPostProcessHook.php";s:4:"796b";s:40:"Classes/Service/CommunicationService.php";s:4:"b428";s:53:"Classes/Service/CommunicationServiceHookInterface.php";s:4:"94c0";s:43:"Classes/Service/ExtensionSettingService.php";s:4:"68f4";s:34:"Classes/Service/LoggingService.php";s:4:"be91";s:35:"Classes/Service/TsConfigService.php";s:4:"bc1a";s:26:"Documentation/Includes.txt";s:4:"cd93";s:23:"Documentation/Index.rst";s:4:"68e5";s:26:"Documentation/Settings.yml";s:4:"3727";s:25:"Documentation/default.vcl";s:4:"cb01";s:38:"Documentation/Administration/Index.rst";s:4:"ae09";s:61:"Documentation/Administration/ExtensionConfiguration/Index.rst";s:4:"8589";s:51:"Documentation/Administration/Installation/Index.rst";s:4:"ecdc";s:45:"Documentation/Administration/PageTs/Index.rst";s:4:"0f1e";s:59:"Documentation/Administration/VarnishConfiguration/Index.rst";s:4:"8b74";s:64:"Documentation/Images/Introduction/Screenshots/IconInEditView.png";s:4:"71dc";s:63:"Documentation/Images/Introduction/Screenshots/IconInWebView.png";s:4:"9bbd";s:58:"Documentation/Images/Introduction/Screenshots/Messages.png";s:4:"2e60";s:36:"Documentation/Introduction/Index.rst";s:4:"3384";s:48:"Documentation/Introduction/Screenshots/Index.rst";s:4:"3140";s:44:"Documentation/Introduction/Support/Index.rst";s:4:"bf43";s:49:"Documentation/Introduction/WhatDoesItDo/Index.rst";s:4:"6f61";s:36:"Documentation/Requirements/Index.rst";s:4:"11ce";s:38:"Resources/Public/Icons/CachePlugin.png";s:4:"4312";s:14:"doc/manual.sxw";s:4:"b4b2";}',
);

